### PR TITLE
Remove usage of deprecated MapSettings

### LINKED ISSUE #3477: Remove deprecated APIs: org.sonar.api.config.Settings [test only]
This is used only in tests.

org.sonar.api.config.Settings

AbstractProjectConfigurationTest

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #3477

### CODE REVIEW

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java [hunk 1]
```diff
@@ -169,36 +147,70 @@ public void giveWarningsWhenGettingProtobufPathAndFolderIsEmpty() throws IOExcep
...
+    return path;
+  }
+
+  private void mockProtobufOutPaths(Configuration configuration) throws IOException{
+    Path path1 = createProtobufOut("report1");
+    Path path2 = createProtobufOut("report2");
+    when(configuration.getStringArray("sonar.cs.analyzer.projectOutPaths")).thenReturn(new String[] {path1.toString(), path2.toString()});
+  }
+
+  private Configuration createEmptyConfiguration(){
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:203]: I would name it `createEmptyMockConfiguration` to make it clearer it's a mock

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java [hunk 2]
```diff
@@ -44,110 +43,87 @@
...
-    settings.setProperty("sonar.cs.analyzer.projectOutPaths", new String[] {path1.toString(), "non-existing"});
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.getStringArray("sonar.cs.analyzer.projectOutPaths")).thenReturn(new String[] {path1.toString(), "non-existing"});
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).containsOnly(path1.resolve("output-cs"));
     assertThat(logTester.logs(LoggerLevel.WARN)).hasSize(1);
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:106]: I know it's not related to your change, but clean as you code 
I think you can replace `hasSize(1)` and `get(0).matches` with `hasOnlyOneElementSatisfying`

Comment (andrei-epure-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:106]: same for lines 117-118; 143-144;

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java [hunk 3]
```diff
@@ -44,110 +43,87 @@
...
   public void informHowManyProtoFilesAreFound() throws IOException {
-    setProtobufOut();
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+    mockProtobufOutPaths(configuration);
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isNotEmpty();
     assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(2);
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:128]: should we also check the `DEBUG` messages? 
also, for the WARN below, should it rather be `hasOnlyOneElementSatisfying` or does it have more messages? Do we know how many?

Comment (andrei-epure-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:128]: this test confuses me together with `onlyProtobufReportsPresent` - the mock has 2 files only - why is it logging that it doesn't find something?

I'm trying to understand the tests in order to see if something is missing

Comment (costin-zaharia-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:128]: This test is misleading, `logTester.logs(LoggerLevel.WARN)` is actually empty (there is no warning). I will change the assert to clarify that.

Comment (andrei-epure-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java]: > Regarding onlyProtobufReportsPresent, I'm not sure what you mean. There are no logs verified in it.

it does the exact same actions as `informHowManyProtoFilesAreFound` . I meant the 2 tests that are testing the same behavior (in different ways) are confusing.

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java [hunk 4]
```diff
@@ -98,35 +98,36 @@ public void noWarningsWhenGettingProtobufPathAndNoPropertyAvailable_TestProject(
...
   @Test
   public void informHowManyProtoFilesAreFound() throws IOException {
-    Configuration configuration = createEmptyConfiguration();
+    Configuration configuration = createEmptyMockConfiguration();
     mockProtobufOutPaths(configuration);
 
     AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isNotEmpty();
-    assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(2);
-    assertThat(logTester.logs(LoggerLevel.WARN)).allMatch(s -> s.endsWith("contains no .pb file(s). No protobuf files will be loaded from this directory."));
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java:153]: lol, talk about test code quality :)) 

indeed, as it was empty, all were matching

**Pull request merged**